resources/frontend_client/index_template.html

```diff
@@ -16,23 +16,34 @@
     <meta name="msapplication-config" content="/app/assets/img/browserconfig.xml">
     <meta name="theme-color" content="#ffffff">
     <meta name="apple-mobile-web-app-status-bar-style" content="default" />
+    <meta name="base-href" content="{{{baseHref}}}" />
+    <meta name="uri" content="{{{uri}}}" />
 
-    {{{embed_code}}}
+    {{{embedCode}}}
 
     <title>Metabase</title>
 
-    <base href={{{base_href}}} />
+    <base href="{{{baseHref}}}" />
 
+    <script type="application/json" id="_metabaseBootstrap">
```

Comment:
I can't recall, is there a reason we don't just use `<meta>` tags for bootstrap/localization as well? It seems like that would be even safer.

Reply:
i don't think it's safer to put JSON in an attribute of a `<meta>` tag vs as the content of a `<script type="application/json">` but I think it looks crazy having big potentially multi-line attributes

Reply:
It is safer because if we escape HTML entities (which includes `<` etc) there's no chance an attacker can inject a `</script>`

Reply:
mentioned this already in Slack but we can't escape HTML entities on the public settings because it busts all the types (which have slashes in them) and stuff like that

Reply:
We could call `unescape` on it, like `JSON.parse(unescape(document.getElementById("_metabaseBootstrap").content));`

The escaping was only really a problem in `<script type="javascript">` tags where we didn't have an opportunity to transform the string before it was executed.